hardware-crazyflie2.md

```diff
@@ -0,0 +1,65 @@
+# Crazyflie 2.0
+
+The Crazyflie line of micro quads was created by Bitcraze AB. An overview of the Crazyflie 2 (CF2) is here: https://www.bitcraze.io/crazyflie-2/
+
+![](images/hardware/hardware-crazyflie2.png)
+
+## Quick Summary
+
+<aside class="tip">
+The main hardware documentation is here: https://wiki.bitcraze.io/projects:crazyflie2:index
+</aside>
+
+  * Main System-on-Chip: STM32F405RG
+    * CPU: 168 MHz ARM Cortex M4 with single-precision FPU
+    * RAM: 192 KB SRAM
+  * nRF51822 radio and power management MCU
+  * MPU9250 Accel / Gyro / Mag
+  * LPS25H barometer
+
+
+## Flashing
+
+After setting up the PX4 development environment, follow these steps to put the PX4 software on the CF2:
+
+1. Grab source code of the PX4 [Bootloader](https://github.com/PX4/Bootloader)
+
+2. Compile using `make crazyflie_bl`
+
+3. Put the CF2 into DFU mode:
+	- Ensure it is initially unpowered
+	- Hold down button
+	- Plug into computer's USB port
+	- After a second, the blue LED should start blinking and after 5 seconds should start blinking faster
+	- Release button
+
+4. Flash bootloader using dfu-util: `sudo dfu-util -d 0483:df11 -a 0 -s 0x08000000 -D crazyflie_bl.bin` and unplug CF2 when done
+	- If successful, then the yellow LED should blink when plugging in again
+
+5. Grab the [Firmware](https://github.com/PX4/Bootloader)
+
+6. Compile with `make crazyflie_default upload`
+
+7. When prompted to plug in device, plug in CF2: the yellow LED should start blinking indicating bootloader mode. Then the red LED should turn on indicating that the flashing process has started.
+
+8. Wait for completion
+
+
+## Usage
+
+The power button has the same functionality as with the stock firmware.
+
+Once the right yellow LED has stopped blinking and the bottom left blue light has started blinking, the boot sequence is complete.
+
+Connect via USB to QGC to finish configuration:
+- Note: Do not use the Airframe tab of QGC
+- Manually set `SYS_AUTOSTART` to 4900 in the parameter options and `SYS_AUTOCONFIG` to 1 (and then restart).
```

Comment:
It sounds like we should auto-detect the board instead. See https://github.com/PX4/Firmware/blob/master/ROMFS/px4fmu_common/init.d/rcS#L158. It would be better to automatically set the SYS_AUTOSTART value and SYS_AUTOCONFIG in the startup script if the board config is detected.

Then we can scrap the entire section.